hudi-common/src/test/java/org/apache/hudi/metadata/TestFileSystemBackedTableMetadata.java

```diff
@@ -0,0 +1,181 @@
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllPartitionPaths().size() == 0);
+  }
+
+  @Test
+  public void testOneLevelPartitionedTableWithoutHoodiePartitionMetaFile() throws IOException {
+    String instant = "100";
+    // Generate 10 files under each partition
+    ONE_LEVEL_PARTITIONS.stream().map(p -> createPartition(p, true))
+        .forEach(p -> createDataFiles(basePath + "/" + p, instant));
+    FileCreateUtils.createCommit(basePath, instant);
+    HoodieLocalEngineContext localEngineContext = new HoodieLocalEngineContext(metaClient.getHadoopConf());
+    FileSystemBackedTableMetadata fileSystemBackedTableMetadata =
+        new FileSystemBackedTableMetadata(localEngineContext, new SerializableConfiguration(metaClient.getHadoopConf()), basePath, false);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllPartitionPaths().size() == 3);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllFilesInPartition(new Path(basePath + "/" + ONE_LEVEL_PARTITIONS.get(0))).length == 10);
+  }
+
+  @Test
+  public void testMultiLevelPartitionedTable() throws IOException {
+    String instant = "100";
+    // Generate 10 files under each partition
+    MULTI_LEVEL_PARTITIONS.stream().map(p -> createPartition(p, true))
+        .forEach(p -> createDataFiles(basePath + "/" + p, instant));
+    FileCreateUtils.createCommit(basePath, instant);
+    HoodieLocalEngineContext localEngineContext = new HoodieLocalEngineContext(metaClient.getHadoopConf());
+    FileSystemBackedTableMetadata fileSystemBackedTableMetadata =
+        new FileSystemBackedTableMetadata(localEngineContext, new SerializableConfiguration(metaClient.getHadoopConf()), basePath, false);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllPartitionPaths().size() == 3);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllFilesInPartition(new Path(basePath + "/" + MULTI_LEVEL_PARTITIONS.get(0))).length == 10);
+  }
+
+  @Test
+  public void testMultiLevelEmptyPartitionTable() throws IOException {
+    String instant = "100";
+    // Generate 10 files under each partition
+    MULTI_LEVEL_PARTITIONS.stream().forEach(p -> createPartition(p, true));
+    FileCreateUtils.createCommit(basePath, instant);
+    HoodieLocalEngineContext localEngineContext = new HoodieLocalEngineContext(metaClient.getHadoopConf());
+    FileSystemBackedTableMetadata fileSystemBackedTableMetadata =
+        new FileSystemBackedTableMetadata(localEngineContext, new SerializableConfiguration(metaClient.getHadoopConf()), basePath, false);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllPartitionPaths().size() == 3);
+    Assertions.assertTrue(fileSystemBackedTableMetadata.getAllFilesInPartition(new Path(basePath + "/" + MULTI_LEVEL_PARTITIONS.get(0))).length == 0);
+  }
+
+  private void createDataFiles(String fullPath, String instant) {
```

Comment:
can we please use the `HoodieTestUtils` methods? We spend a lot of time chasing after these one-offs.